### ISSUE TITLE
instanceof breaks open declaration

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-jdt/eclipse.jdt.core#853]
Using latest master but also with 4.25.

Per occasion stumbled over similar code and wanted to navigate via Ctrl+Click into `CompletableFuture.supplyAsync()` call (or just via F3 with cursor in this code) => doesn't work, no terror, no feedback, just no navigation.

This seem to be related to the "instanceof" check below - if removed, navigation to `CompletableFuture.supplyAsync()` works again.

```
import java.util.concurrent.CompletableFuture;

public class F3NotWorking {
    public CompletableFuture<String> readOutStream() {
        return CompletableFuture.supplyAsync(() -> { // <-- F3 to open supplyAsync() declaration doesn't work
        	try {
                return "";
            } catch (Throwable e) {
                String message = "";
                if (e instanceof Exception) { // replace with message == "" and F3 will work again
                    message = "This instanceof breaks F3 above!!!";
                }
                throw new RuntimeException(message, e);
            }
        });
    }
}
```

**Comment by trancexpress** [comment]
Fixed by #849 so must be a duplicate of #848.

**Comment by iloveeclipse** [comment]
Closing as duplicate of https://github.com/eclipse-jdt/eclipse.jdt.core/issues/848.
